### PR TITLE
Csvup 8 feature keys

### PR DESCRIPTION
**Pull request by dklein1211**

### CODE REVIEW

**Comment by AnthonyPerish** [comment]
Feature keys bug: If I set a feature key, then change the feature key to none, an error is thrown when trying to save.

Screen Shot 2019-09-06 at 10 37 38 AM

Screen Shot 2019-09-06 at 10 38 36 AM

Screen Shot 2019-09-06 at 10 38 57 AM

**Comment by AnthonyPerish** [comment]
Feature Keys Bug: Device update API is being called twice for each device which often times results in a datastore conflict due to updating the device twice in a row too quick.  This seems to happen every time for me when I have multiple users/devices.  Screenshots attached are from clicking proceed once. 

Screen Shot 2019-09-06 at 10 55 54 AM

Screen Shot 2019-09-06 at 10 56 14 AM

**Comment by AnthonyPerish** [comment]
On initial screen upon loading app the "upload a CSV File with the following columns" text needs to be updated to include the new column softphone(yes|no).

**Comment by AnthonyPerish** [comment]
When trying to create a user with no device, the following error is thrown about the brand, plus the proceed button is disabled.  In the instance when nothing is specified for a device this should not happen.

Screen Shot 2019-09-06 at 11 25 30 AM

File: app.js [hunk 1]
```diff
@@ -486,7 +475,250 @@ define(function(require) {
...
+		},
+
+		getDeviceList: function(device, deviceList) {
+			if (!deviceList.length) {
+				deviceList.push(device);
+			} else {
+				//Verify if the targeted device (deviceInfo) has been added to the device list (deviceModels).
+				//This is used narrow down the API calls to the provisioner to get the max feature keys that device can have.
+				_.each(deviceList, function(listDevice) {
+					var found = false;
```
Comment (AnthonyPerish) [inline app.js]: This whole _.each section seems unnecessarily complex.  Seems you could replace it with if (device.brand !== listDevice.brand || device.family !== listDevice.family || device.model !== listDevice.model { deviceList.push(device); }

Comment (dklein1211) [reply app.js]: This section has been removed in my latest commit and simplified.

**Comment by dklein1211** [comment]
> When trying to create a user with no device, the following error is thrown about the brand, plus the proceed button is disabled. In the instance when nothing is specified for a device this should not happen.
> 
> Screen Shot 2019-09-06 at 11 25 30 AM

**Comment by dklein1211** [comment]
> Feature keys bug: If I set a feature key, then change the feature key to none, an error is thrown when trying to save.
> 
> Screen Shot 2019-09-06 at 10 37 38 AM
> 
> Screen Shot 2019-09-06 at 10 38 36 AM
> 
> Screen Shot 2019-09-06 at 10 38 57 AM

This has been resolved in the current commit

**Pull request merged**